packages/builder/src/builtin-tasks/run.ts

```diff
@@ -17,17 +22,29 @@ function filterNonExistent (scripts: string[]): string[] {
 }
 
 export async function runMultipleScripts (runtimeEnv: AlgobRuntimeEnv,
-  scriptNames: string[],
-  log: (...args: unknown[]) => unknown): Promise<void> {
+                                          scriptNames: string[],
+                                          log: (...args: unknown[]) => unknown,
+                                          persistSnapshots: boolean,
```

Comment:
we are not doing a `snaphost` --> snapshot means that we serialize the current state. We do a checkpoint here.
 So let's rename to `doCheckpoint`

Reply:
Is this simply a rename of `persistSnapshots` to `doCheckpoint`?

Reply:
yes

Reply:
Created a `Deployer` class and this logic gets handled inside of each task.